src/Physics/HELepton/EventGen/PhotonRESWdecPythia6.cxx

```diff
@@ -0,0 +1,326 @@
+      p4longo.BoostZ(beta);
+
+      TLorentzVector p4o( (double)p4longo.Px(), (double)p4longo.Py(), (double)p4longo.Pz(), (double)p4longo.E() );
+      p4o.RotateUz(unit_nu);
+
+      TParticlePDG * part = PDGLibrary::Instance()->Find(pdgc);
+      if ( (ks==1 || ks==4) && p4o.E() < part->Mass() ) {
+        LOG("PhotonRESGenerator", pWARN) << "Putting at rest one stable particle generated by PYTHIA because E < m";
+        LOG("PhotonRESGenerator", pWARN) << "PDG = " << pdgc << " // State = " << ks;
+        LOG("PhotonRESGenerator", pWARN) << "E = " << p4o.E() << " // |p| = " << TMath::Sqrt(p4o.P());
+        LOG("PhotonRESGenerator", pWARN) << "p = [ " << p4o.Px() << " , "  << p4o.Py() << " , "  << p4o.Pz() << " ]";
+        LOG("PhotonRESGenerator", pWARN) << "m    = " << p4o.M() << " // mpdg = " << part->Mass();
+        p4o.SetXYZT(0,0,0,part->Mass());
+      }
+
+      // copy final state particles to the event record
+      GHepStatus_t ist = (ks==1 || ks==4) ? kIStStableFinalState : kIStDISPreFragmHadronicState;
+
+      // fix numbering scheme used for mother/daughter assignments
+      int firstmother = -1;
+      int lastmother  = -1;
+      int firstchild  = -1;
+      int lastchild   = -1;
+
+      if ( particle->GetParent() < 10 ) {
+        if      ( TMath::Abs(pdgc)<7 ) {   //outgoing quarks: mother will be the boson (saved in position 4)
+          firstmother = 4;
+          firstchild  = particle->GetFirstChild() - 6;
+          lastchild   = particle->GetLastChild()  - 6;
+        }
+        else if ( TMath::Abs(pdgc)==24 ) { //produced W boson: mother will be the incoming neutrino
+          firstmother = 0;
+          firstchild  = particle->GetFirstChild() - 6;
+          lastchild   = particle->GetLastChild()  - 6;
+        }
+        else if ( pdgc==22 ) {             //radiative photons: mother will be the incoming electron
+          firstmother = 2;
+        }
+      }
+      else { //rest
+        firstmother = particle->GetParent()     - 6; //shift to match boson position
+        firstchild  = (particle->GetFirstChild()==0) ? particle->GetFirstChild() - 1 : particle->GetFirstChild() - 6;
+        lastchild   = (particle->GetLastChild()==0)  ? particle->GetLastChild()  - 1 : particle->GetLastChild()  - 6;
+      }
+
+      double lightspeed = 299792458e3; //c in mm/s. Used for time in PYTHIA t[s]=t_pythia[mm]/c[mm/s]
+      double vx = nu->X4()->X() + particle->GetVx()*1e12; //pythia gives position in [mm] while genie uses [fm]
+      double vy = nu->X4()->Y() + particle->GetVy()*1e12;
+      double vz = nu->X4()->Z() + particle->GetVz()*1e12;
+      double vt = nu->X4()->T() + particle->GetTime()/lightspeed;
+      TLorentzVector pos( vx, vy, vz, vt );
+
+      event->AddParticle(pdgc, ist, firstmother, lastmother, firstchild, lastchild, p4o, pos );
+
+    }
+
+    delete particle;
+    pythia_particles->Clear("C");
```

Comment:
Just checking here.  The `delete particle` would delete just the last entry of the `pythia_particles` list, would it not?  I'm not entirely clear on the ownership model behind all this, but would like assurances that the call to `Clear("C")` doesn't cause a double deletion.  Because there's nothing intervening, I'm not sure anything bad would happen if that were the case, but it best not to have that happen.   The alternative possibility is that it should be deleting each particle as it's iterating and we have a potential memory leak.  I'm not sure what the `Clear()` is actually doing.